### ISSUE TITLE
Comet output not going to output-dir

### ISSUE THREAD

**Issue by mriffle** [crux-toolkit/crux-toolkit#483]
Using the most recent build (crux-3.2.583fa5c.Linux.x86_64.zip), the output from comet is not going into crux-output. And it also ignores an output directory specified by --output-dir  A crux-output directory is created and a log and comet.params.txt are put in there; however, all the output of comet (pin, pep xml, etc) are placed in the current working directory.

**Comment by CharlesEGrant** [comment]
Hi Michael,

I haven't been able to reproduce this problem. For example, this is what I get on my Linux box when I run comet on the included samples, creating an output directory named 'foobar':

[cegrant@frege-linux crux-3.2.Linux.x86_64]$ bin/crux comet --output-dir foobar doc/example-files/demo.ms2 doc/example-files/small-yeast.fasta
INFO: Writing results to output directory 'foobar'.
INFO: CPU: frege-linux
INFO: Crux version: 3.2-583fa5c
INFO: Mon Jan  4 20:08:26 PST 2021
INFO: Beginning comet.
INFO:  Comet version "2019.01 rev. 5"
INFO:
INFO:  Search start:  01/04/2021, 08:08:26 PM
INFO:  - Input file: doc/example-files/demo.ms2
INFO:    - Load spectra: 166
INFO:      - Search progress: 100%
INFO:      - Post analysis:  done
INFO:  Search end:    01/04/2021, 08:08:26 PM, 0m:0s
INFO:
INFO: Elapsed time: 0.1 s
INFO: Finished crux comet.
INFO: Return Code:0
[cegrant@frege-linux crux-3.2.Linux.x86_64]$ ls foobar/
comet.log.txt  comet.params.txt  comet.target.pep.xml  comet.target.txt

Which seems to the be correct behavior. Could you forward us a copy of any parameter file you are using and the log file?

**Comment by mriffle** [comment]
Hi Charles,

I'm not using a params file. Here's what I'm typing and the results:

```
mriffle@imp1:~/work/crux_comet_run_multi$ ~/work/crux/crux-3.2.Linux.x86_64/bin/crux comet --output-dir foobar --decoy_search 1 --output_percolatorfile 1 ./UWPRLumos_2019_0605_AZ_013_eh237_AZ.mzML ./UWPRLumos_2019_0605_AZ_016_eh238_AZ.mzML ./2019-06-uniprot-proteome-cow-3AUP000009136_contam_MIND-OA-Dam1-Arp23.fasta
INFO: Writing results to output directory 'foobar'.
INFO: CPU: imp1
INFO: Crux version: 3.2-583fa5c
INFO: Mon Jan  4 21:09:51 PST 2021
INFO: Beginning comet.
INFO:  Comet version "2019.01 rev. 5"
INFO:
INFO:  Search start:  01/04/2021, 09:09:51 PM
INFO:  - Input file: ./UWPRLumos_2019_0605_AZ_013_eh237_AZ.mzML
INFO:    - Load spectra: 52831
INFO:      - Search progress:  99%
INFO:      - Post analysis:  done
INFO:  Search end:    01/04/2021, 09:10:55 PM, 1m:4s
INFO:
INFO:  Search start:  01/04/2021, 09:10:55 PM
INFO:  - Input file: ./UWPRLumos_2019_0605_AZ_016_eh238_AZ.mzML
INFO:    - Load spectra: 51041
INFO:      - Search progress:  99%
INFO:      - Post analysis:  done
INFO:  Search end:    01/04/2021, 09:12:05 PM, 1m:10s
INFO:
INFO: Elapsed time: 134 s
INFO: Finished crux comet.
INFO: Return Code:0
```

```
mriffle@imp1:~/work/crux_comet_run_multi$ ls
2019-06-uniprot-proteome-cow-3AUP000009136_contam_MIND-OA-Dam1-Arp23.fasta  UWPRLumos_2019_0605_AZ_013_eh237_AZ.target.pep.xml  UWPRLumos_2019_0605_AZ_016_eh238_AZ.pin             arp23-index                              foobar
UWPRLumos_2019_0605_AZ_013_eh237_AZ.mzML                                    UWPRLumos_2019_0605_AZ_013_eh237_AZ.target.txt      UWPRLumos_2019_0605_AZ_016_eh238_AZ.target.pep.xml  crux-comet-perc-subgroups.limelight.xml
UWPRLumos_2019_0605_AZ_013_eh237_AZ.pin                                     UWPRLumos_2019_0605_AZ_016_eh238_AZ.mzML            UWPRLumos_2019_0605_AZ_016_eh238_AZ.target.txt      crux3.2-output
```

```
mriffle@imp1:~/work/crux_comet_run_multi$ ls foobar
comet.log.txt  comet.params.txt
```


Here's the log:

```
mriffle@imp1:~/work/crux_comet_run_multi/foobar$ cat comet.log.txt
INFO: CPU: imp1
INFO: Crux version: 3.2-583fa5c
INFO: Mon Jan  4 21:09:51 PST 2021
COMMAND LINE: /home/mriffle/work/crux/crux-3.2.Linux.x86_64/bin/crux comet --output-dir foobar --decoy_search 1 --output_percolatorfile 1 ./UWPRLumos_2019_0605_AZ_013_eh237_AZ.mzML ./UWPRLumos_2019_0605_AZ_016_eh238_AZ.mzML ./2019-06-uniprot-proteome-cow-3AUP000009136_contam_MIND-OA-Dam1-Arp23.fasta
INFO: Beginning comet.
INFO:  Comet version "2019.01 rev. 5"
INFO:
INFO:  Search start:  01/04/2021, 09:09:51 PM
INFO:  - Input file: ./UWPRLumos_2019_0605_AZ_013_eh237_AZ.mzML
INFO:    - Load spectra: 52831
INFO:      - Search progress:  99%
INFO:      - Post analysis:  done
INFO:  Search end:    01/04/2021, 09:10:55 PM, 1m:4s
INFO:
INFO:  Search start:  01/04/2021, 09:10:55 PM
INFO:  - Input file: ./UWPRLumos_2019_0605_AZ_016_eh238_AZ.mzML
INFO:    - Load spectra: 51041
INFO:      - Search progress:  99%
INFO:      - Post analysis:  done
INFO:  Search end:    01/04/2021, 09:12:05 PM, 1m:10s
INFO:
INFO: Elapsed time: 134 s
INFO: Finished crux comet.
INFO: Return Code:0
```

**Comment by CharlesEGrant** [comment]
Ah! I can reproduce this now. The bug only surfaces when multiple spectra files are specified. I should be able to sort this out fairly quickly.

**Comment by CharlesEGrant** [comment]
Fixed as part of update to Comet 2021-02-0 (e35012f4f765cbdce21a6dbc85240de0076b2784).
